File: packages/restapi/src/lib/chat/helpers/wallet.ts
import { Signer, pCAIP10ToWallet } from '../../helpers';
import { SignerType, walletType } from '../../types';

export const getWallet = (
  options: walletType
): {
  account: string | null;
  signer: SignerType | null;
} => {
  const { account, signer } = options || {};

  return {
    account: account ? pCAIP10ToWallet(account) : account,
    signer,
  };
};

export const getAccountAddress = async (
  options: walletType
): Promise<string> => {
  const { account, signer } = options || {};

  const pushSigner = new Signer(signer as SignerType);
  return account || (await pushSigner.getAddress()) || '';
};

File: packages/push-api-dart/lib/src/helpers/src/wallet.dart
import '../../../push_api_dart.dart';

Wallet getWallet({
  String? address,
  Signer? signer,
}) {
  return Wallet(
    signer: signer,
    address: address != null ? pCAIP10ToWallet(address) : address,
  );
}

String getAccountAddress(Wallet wallet) {
  final address = wallet.address ?? wallet.signer?.getAddress();
  if (address == null) {
    throw Exception('Address not found');
  } else {
    return address;
  }
}
